### DISCUSSION TITLE
Help on long running tasks

### DISCUSSION THREAD

**Discussion by 5u623l20** [cirruslabs/cirrus-ci-docs#1007] [Q&A]
Currently we can exceed the 120m time limit if either of the following conditions are met:
  - Persistent Worker
  - Compute Service
  - Compute Credits

Is there any way to add a single logic to a task that it will be executed only if either of the above is true. I know it also creates a race condition as what if two of the conditions are met and where is the task supposed to run?

I can create three different tasks and keep it in the file but the problem is the URL of the artifacts will change hence it is difficult to integrate in a dashboard.

**Comment by fkorotkov** [comment]
Could you please elaborate your use case a bit? If the problem only in the artifacts URLs then you might be interested in [`alias`es](https://cirrus-ci.org/guide/writing-tasks/#task-execution-dependencies) which can be used in URLs to artifacts. If all the tasks will have the same alias then the URL prefix will be the same for them.

**Reply by 5u623l20** [reply]
We have couple of tasks on different architectures. The build times are fine which are under 90m actually but the problem is with the tests. Those takes long hours around 6-8 hours for non-x86 ARCHS which we want to run in any of three mentioned above.

[alias](https://www.freshports.org/security/py-cryptography/) looks like a good option. I have to add more logic to task depends and only_ifs because in case more than two options are defined two same tasks are going to be executed and writing in the same artifact location. I am not sure about the behavior on what happens when there are same filenames of same artifacts although it says it's saved under a zip file but I will have to check in details.
